ambry-api/src/main/java/com.github.ambry/clustermap/ClusterAgentsFactory.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.clustermap;
+
+import java.io.IOException;
+
+
+/**
+ * A factory interface to get cluster agents such as {@link ClusterMap} and {@link ClusterParticipant}
+ */
+public interface ClusterAgentsFactory {
+  /**
+   * Get a reference to a {@link ClusterMap}
+   */
+  ClusterMap getClusterMap() throws IOException;
+
+  /**
+   * Get a reference to a {@link ClusterParticipant}
+   */
+  ClusterParticipant getClusterParticipant() throws IOException;
```

Comment:
`participant` is a generic enough term in distributed systems and protocols. It is used to represent a constituent entity that has an active part in the workings of the distributed system. Take the [two-phase commit protocol](https://en.wikipedia.org/wiki/Two-phase_commit_protocol), for instance.
I am using the term "agent" to represent any cluster component, including both participants and the cluster manager (which is a listener).